browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc

```diff
@@ -413,10 +442,19 @@ void BraveNewTabMessageHandler::HandleSaveNewTabPagePref(
   const auto settingsValueBool = settingsValue.GetBool();
   if (settingsKeyInput == "showBackgroundImage") {
     settingsKey = kNewTabPageShowBackgroundImage;
+    bool is_sponsored_image_enabled =
+        settingsValueBool &&
+        prefs->GetBoolean(kNewTabPageShowSponsoredImagesBackgroundImage);
+    UMA_HISTOGRAM_BOOLEAN("Brave.NTP.SponsoredImagesEnabled",
```

Comment:
it is preferable to emit histogram from one place, so maybe we could wrap this into a helper function https://chromium.googlesource.com/chromium/src.git/+/HEAD/tools/metrics/histograms#coding-emitting-to-histograms

Reply:
I'm curious whether we should inject these here and in appearance handler? Wouldn't changing the pref here trigger BraveAppearanceHandler::OnBackgroundPreferenceChanged

Reply:
I'm not sure if `BraveAppearanceHandler` (used on brave://settings, via the polymer pages) is registered when we're on the NTP. I'll check to see

Reply:
It isn't. But you could extract this part that both would use to a new MessageHandler, or just use an independent helper function.

Reply:
Fixed!